src/decoder/mod.rs

```diff
@@ -67,16 +67,41 @@ pub struct Decoder<R: Read> {
     limits: Limits,
 }
 
-struct InterlacedRow<'data> {
-    data: &'data [u8],
-    interlace: InterlaceInfo,
+/// A row of data with interlace information attached.
+#[derive(Debug)]
+#[non_exhaustive]
+pub struct InterlacedRow<'data> {
+    pub data: &'data [u8],
+    pub interlace: InterlaceInfo,
 }
 
-enum InterlaceInfo {
-    None,
+/// PNG (2003) specifies two interlace modes, but reserves future extensions.
+#[derive(Debug)]
+#[non_exhaustive]
```

Comment:
In the tradeoff between marking it `Copy` and `#[non_exhaustive]`, I'd choose to have `Copy` etc. derives. The probability that the standard introduces a new interlace method before we bump major version is quite small.